python/llm/src/bigdl/llm/transformers/speculative.py

```diff
@@ -430,14 +457,17 @@ def speculative_generate(self,
 
             if self.device.type == 'cpu':
                 # init past_key_values_storage and assign initial fp32 value
-                if step == 1:
+                if step == 1 and self.config.model_type != "gpt_bigcode":
```

Comment:
instead of checking `self.config.model_type != "gpt_bigcode"` here,  have `_prepare_past_key_values_storage_cpu` to correctly handle `gpt_bigcode` instead

Reply:
Yes, WIP on it